File: GuessWordGame/GuessWordGame/Task.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GuessWordGame
{
    public struct Task
    {
        public string Question;
        public string Answer;
        public Task(string question, string answer)
        {
            Question = question;
            Answer = answer;
        }
    }
}

File: Task.ts
import data from '../tasks/all.json';

interface TaskData {
	question: string;
	answer: string;
}

class Task implements TaskData {
	question: string;
	answer: string;

	constructor(data: TaskData) {
		this.question = data.question;
		this.answer = data.answer;
	}
}

function isTaskData(arg: any): arg is TaskData {
	return arg && !!arg.question && !!arg.answer;
}

const tasks: Task[] = [];

for (const record of data) {
	if (!isTaskData(record)) { continue; }
	tasks.push(new Task(record));
}
console.log(tasks);
